iceoryx_utils/include/iceoryx_utils/internal/cxx/vector.inl

```diff
@@ -155,11 +155,11 @@ inline uint64_t vector<T, Capacity>::capacity() const
 template <typename T, uint64_t Capacity>
 inline void vector<T, Capacity>::clear()
 {
-    for (uint64_t i = 0u; i < m_size; ++i)
+    for (uint64_t i = m_size; i != 0U; --i)
     {
-        at(i).~T();
+        at(i - 1).~T();
```

Comment:
For efficiency it would be better to start at m_size - 1. But care must be taken that the element at position zero is destroyed and the loop left correctly (i is unsigned and will wrap to max). A while loop might be needed then, but I would try to avoid duplicate decrements. It is not clear that they are optimized away...

Reply:
ideally `while(pob_back()) {}` could be used, if `pop_back()` would return a `bool`. Either the signature of `pop_back()` could be adjusted or a `pop_back_internal` with that signature could be used ... or (don't take it serious) `while(pop_back(), m_size > 0) {}`